components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java

```diff
@@ -126,6 +128,36 @@ public void setPasswordFinder(Supplier<char[]> passwordFinder) {
             InputStreamReader isr = null;
             InputStream is = null;
             try {
+                is = ResourceHelper.resolveMandatoryResourceAsInputStream(camelContext, resource);
```

Comment:
LGTM to me, I would use a try-with-resource here nevertheless

Reply:
This is the same line as for "normal" path, where `is` is closed using `IoUtils.closeQuietly(r, is, isr)` in finally block.
`org.apache.sshd.common.util.security.SecurityUtils#loadKeyPairIdentities()` consumes the stream, so there are two such lines.

Reply:
Ah yeah, didn't notice that. All good then. Merge if you want.

Reply:
let's wait for CI.